### DISCUSSION TITLE
Error en variables sin usar en Maven

### DISCUSSION THREAD

**Discussion by mbeorlegui** [dds-jv/foro#361] [Consultas de herramientas] [answered]
Nuevamente Maven me bardea cuando hay variables que tengo declaradas y no uso (**pero segun la consigna necesito**), y tengo miedo de que a la hora de entregar el trabajo no me validen los tests que realizan desde Actions, que según vi, es lo que se hace.

No se si hay alguna opcion de cambiar esto desde el código, o simplemente de ignorar estos errores.

El error que me tira es el siguiente:
```java
[ERROR] Medium: Unread field: domain.miembro.Miembro.apellido [domain.miembro.Miembro] At Miembro.java:[line 20] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.miembro.Miembro.nombre [domain.miembro.Miembro] At Miembro.java:[line 19] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.miembro.Miembro.numeroDeDocumento [domain.miembro.Miembro] At Miembro.java:[line 21] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.miembro.Miembro.tipoDeDocumento [domain.miembro.Miembro] At Miembro.java:[line 22] URF_UNREAD_FIELD
```

Y es por las siguientes lineas de codigo:
```java
public class Miembro {
  private String nombre;
  private String apellido;
  private Integer numeroDeDocumento;
  private Documento tipoDeDocumento;
  private List<Sector> sectores = new ArrayList<>();

  public Miembro(
      String nombre, String apellido, Integer numeroDeDocumento, Documento tipoDeDocumento) {
    this.nombre = nombre;
    this.apellido = apellido;
    this.numeroDeDocumento = numeroDeDocumento;
    this.tipoDeDocumento = tipoDeDocumento;
  }
}
```

Otra solucion que me tira el IntelliJ es la siguiente: `Convert field to local variable in constructor`

Que al parecer lo único que hace, luego de aplicarlo para todos las propiedades del constructor, es dejarlo sólo como argumento del mismo, cosa que creo que estaría mal.

Quedaría de la siguiente forma:
```java
public class Miembro {
  private List<Sector> sectores = new ArrayList<>();

  public Miembro(
      String nombre, String apellido, Integer numeroDeDocumento, Documento tipoDeDocumento) {
  }
}
```

No sé qué opciones tengo para hacer que esto no me rompa cuando hago el `mvn verify`, espero que me puedan ayudar.

Desde ya muchas gracias.

**Comment by gastonprieto** [comment, marked as answer]
Buenas @mbeorlegui !

Como el TPA es iterativo, es normal que tengas campos que aún no usas, podrías usar @SuppressFBWarnings para ignorar ese error mientras tanto

**Reply by tobiascastaneda2000** [reply]
Buenas noches! Mil disculpas las molestias, pero a mi me apareció un error muy parecido.

Al hacer el mvn test, por un lado te deja correr los tests, pero si hago mvn verify me rompe todo la copilacion con el error muy similar a tuyo.

Paso capturas por las dudas:

[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ ejercicio ---
[INFO] Building jar: C:\Disenio\TP_ANUAL_1\2022-tpa-ju-ma-grupo-04\target\ejercicio-1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.2:check (Validar Formateo) @ ejercicio ---
[WARNING] Old version of checkstyle detected. Consider updating to >= v8.30
[WARNING] For more information see: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html
[INFO] Ignored 11 errors, 0 violation remaining.
[INFO] You have 0 Checkstyle violations.
[INFO]
[INFO] >>> spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) > :spotbugs @ ejercicio >>>
[INFO]
[INFO] --- spotbugs-maven-plugin:4.2.0:spotbugs (spotbugs) @ ejercicio ---
[INFO] Fork Value is true
[INFO] Done SpotBugs Analysis....
[INFO] 
[INFO] <<< spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) < :spotbugs @ ejercicio <<<
[INFO]
[INFO]
[INFO] --- spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) @ ejercicio ---
[INFO] BugInstance size is 33
[INFO] Error size is 0
[INFO] Total bugs: 33
[ERROR] Medium: Unread field: domain.organizacion.Organizacion.razonSocial [domain.organizacion.Organizacion] At Organizacion.java:[line 16] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.Organizacion.tipo [domain.organizacion.Organizacion] At Organizacion.java:[line 19] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.Organizacion.ubicacionGeografica [domain.organizacion.Organizacion] At Organizacion.java:[line 20] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.Sector.miembroACargo [domain.organizacion.Sector] At Sector.java:[line 12] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.miembro.Miembro.apellido [domain.organizacion.miembro.Miembro] At Miembro.java:[line 20] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.miembro.Miembro.nombre [domain.organizacion.miembro.Miembro] At Miembro.java:[line 19] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.miembro.Miembro.numeroDocumento [domain.organizacion.miembro.Miembro] At Miembro.java:[line 22] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.miembro.Miembro.tipoDocumento [domain.organizacion.miembro.Miembro] At Miembro.java:[line 21] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.organizacion.miembro.Miembro.trayectos [domain.organizacion.miembro.Miembro] At Miembro.java:[line 23] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.Coordenada.latitud [domain.trayecto.Coordenada] At Coordenada.java:[line 8] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.Coordenada.longitud [domain.trayecto.Coordenada] At Coordenada.java:[line 9] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.Trayecto.tramos [domain.trayecto.Trayecto] At Trayecto.java:[line 10] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.lineas.parada.EstacionSubte.nombre [domain.trayecto.especificaciones.lineas.parada.EstacionSubte] At EstacionSubte.java:[line 7
] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.lineas.parada.EstacionTren.nombre [domain.trayecto.especificaciones.lineas.parada.EstacionTren] At EstacionTren.java:[line 7] U
RF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.lineas.parada.ParadaColectivo.coordenadaDeParada [domain.trayecto.especificaciones.lineas.parada.ParadaColectivo] At ParadaCole
ctivo.java:[line 9] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoAPie.lugarFin [domain.trayecto.especificaciones.tramo.TramoAPie] At TramoAPie.java:[line 11] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoAPie.lugarInicio [domain.trayecto.especificaciones.tramo.TramoAPie] At TramoAPie.java:[line 10] URF_UNREAD_FIELD    
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoBicicleta.lugarFin [domain.trayecto.especificaciones.tramo.TramoBicicleta] At TramoBicicleta.java:[line 11] URF_UNRE
AD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoBicicleta.lugarInicio [domain.trayecto.especificaciones.tramo.TramoBicicleta] At TramoBicicleta.java:[line 10] URF_U
NREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoServicioContratado.lugarFin [domain.trayecto.especificaciones.tramo.TramoServicioContratado] At TramoServicioContrat
ado.java:[line 15] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoServicioContratado.lugarInicio [domain.trayecto.especificaciones.tramo.TramoServicioContratado] At TramoServicioCont
ratado.java:[line 14] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoServicioContratado.servicio [domain.trayecto.especificaciones.tramo.TramoServicioContratado] At TramoServicioContrat
ado.java:[line 16] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoTransportePublico.linea [domain.trayecto.especificaciones.tramo.TramoTransportePublico] At TramoTransportePublico.ja
va:[line 16] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoTransportePublico.paradaFin [domain.trayecto.especificaciones.tramo.TramoTransportePublico] At TramoTransportePublic
o.java:[line 18] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoTransportePublico.paradaInicio [domain.trayecto.especificaciones.tramo.TramoTransportePublico] At TramoTransportePub
lico.java:[line 17] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoTransportePublico.tipoVehiculo [domain.trayecto.especificaciones.tramo.TramoTransportePublico] At TramoTransportePub
lico.java:[line 15] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoVehiculoParticular.lugarFin [domain.trayecto.especificaciones.tramo.TramoVehiculoParticular] At TramoVehiculoParticu
lar.java:[line 18] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoVehiculoParticular.lugarInicio [domain.trayecto.especificaciones.tramo.TramoVehiculoParticular] At TramoVehiculoPart
icular.java:[line 17] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoVehiculoParticular.tipoCombustible [domain.trayecto.especificaciones.tramo.TramoVehiculoParticular] At TramoVehiculo
Particular.java:[line 16] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: domain.trayecto.especificaciones.tramo.TramoVehiculoParticular.tipoVehiculo [domain.trayecto.especificaciones.tramo.TramoVehiculoParticular] At TramoVehiculoPar
ticular.java:[line 15] URF_UNREAD_FIELD
[ERROR] Medium: Unread field: seguridad.Cuenta.password [seguridad.Cuenta] At Cuenta.java:[line 9] URF_UNREAD_FIELD
[INFO] Total time:  22.934 s
[INFO] Finished at: 2022-05-12T23:42:54-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) on project ejercicio: failed with 33 bugs and 0 errors  -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.

Es rarisimo por que en la primera entrega del tp a mi grupo nos funciono perfecto, y este error aparecio recien cuando tratamos de retomarlo esta semana.

En que consiste SuppressFBWarnings?

**Reply by mbeorlegui** [reply]
Buenas Tobias, nosotros lo "_solucionamos_" haciéndole getters a esos atributos que no usamos (sí, es una mala práctica hacer getters innecesarios pero no encontramos otra forma porque el SupressWarnings no nos anduvo).

Fijate que podes usar el Code Generator de IntelliJ para que te haga estos Getters automágicamente.

Espero que te sirva al menos para la entrega, saludos.
